src/Microsoft.Health.Dicom.Core/Messages/Partitioning/GetOrAddPartitionRequest.cs

```diff
@@ -7,13 +7,19 @@
 
 namespace Microsoft.Health.Dicom.Core.Messages.Partitioning;
 
-public class GetPartitionRequest : IRequest<GetPartitionResponse>
+public class GetOrAddPartitionRequest : IRequest<GetOrAddPartitionResponse>
 {
-    public GetPartitionRequest(string paritionName)
+    public GetOrAddPartitionRequest(string partitionName, bool addIfNotExists)
     {
-        PartitionName = paritionName;
+        PartitionName = partitionName;
+        AddIfNotExists = addIfNotExists;
     }
 
+    /// <summary>
+    /// Should the request attempt add if the partition doesn't exist.
+    /// </summary>
+    public bool AddIfNotExists { get; set; }
```

Comment:
I'm not 100% in love with this pattern, but couldn't come up with anything better. The goal is that only on some paths do we actually want to do the `AddIfNotExists` path, but it seemed weird to have 2 paths for messages through the code.

Reply:
Let's move to two paths to make sure the auth story stays correct.